// eslint-config-zoe/index.js
"use strict";

module.exports = {
  root: true,

  parser: "babel-eslint",

  plugins: [
    "import",
    "jsx-a11y",
    "react",
    "react-hooks",
    "promise",
    "unicorn",
    "no-use-extend-native",
    "jest",
    "prettier",
  ],

  env: {
    browser: true,
    commonjs: true,
    es6: true,
    jest: true,
    node: true,
  },

  parserOptions: {
    ecmaVersion: 2018,
    sourceType: "module",
    ecmaFeatures: {
      jsx: true,
    },
  },

  settings: {
    react: {
      version: "detect",
    },
  },

  extends: [
    "eslint:recommended",
    "plugin:unicorn/recommended",
    "plugin:promise/recommended",
    "plugin:jest/recommended",
    "plugin:react/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "airbnb",
    "airbnb/hooks",
    "prettier",
    "prettier/react",
    "prettier/unicorn",
  ],

  rules: {
    "no-console": "off",
    "global-require": "off",
    "no-restricted-syntax": "off",

    // eslint-plugin-prettier
    "prettier/prettier": "warn",

    // eslint-plugin-no-use-extend-native
    "no-use-extend-native/no-use-extend-native": "error",

    // eslint-plugin-react
    "react/prop-types": "off",
    "react/jsx-filename-extension": "off",

    // eslint-plugin-unicorn
    "unicorn/filename-case": "off",
    "unicorn/prevent-abbreviations": "off",
    "unicorn/no-abusive-eslint-disable": "off",
    "unicorn/no-unsafe-regex": "error",
  },
};
